File: src/main/java/com/teamwizardry/librarianlib/gui/template/ComponentTemplate.java
package com.teamwizardry.librarianlib.gui.template;

import com.teamwizardry.librarianlib.gui.GuiComponent;

public class ComponentTemplate<T extends GuiComponent<?>> {

	protected T result;
	
	public T get() {
		return result;
	}
	
}

File: src/main/java/com/teamwizardry/librarianlib/client/gui/template/ComponentTemplate.kt
package com.teamwizardry.librarianlib.client.gui.template

import com.teamwizardry.librarianlib.client.gui.GuiComponent

open class ComponentTemplate<T : GuiComponent<*>>(protected val result: T) {

    fun get(): T {
        return result
    }

}
